spotify_player/src/cli/client.rs

```diff
@@ -344,9 +365,435 @@ async fn handle_playback_request(
             };
             PlayerRequest::SeekTrack(progress + chrono::Duration::milliseconds(position_offset_ms))
         }
+        _ => unreachable!(),
     };
 
     client.handle_player_request(state, player_request).await?;
     client.update_playback(state);
     Ok(())
 }
+
+async fn handle_playlist_request(
+    client: &Client,
+    state: &SharedState,
+    command: Command,
+) -> Result<String> {
+    match command {
+        Command::PlaylistNew {
+            name,
+            public,
+            collab,
+            description,
+        } => {
+            let user = state.data.read().user_data.user.to_owned().unwrap();
+            let id = user.id;
+
+            let resp = client
+                .spotify
+                .user_playlist_create(
+                    id,
+                    name.as_str(),
+                    Some(public),
+                    Some(collab),
+                    Some(description.as_str()),
+                )
+                .await?;
+            Ok(format!(
+                "Playlist '{}' with id '{}' was created.",
+                resp.name,
+                resp.id.to_string()
+            ))
+        }
+        Command::PlaylistDelete { id } => {
+            let user = state.data.read().user_data.user.to_owned().unwrap();
+            let uid = user.id;
+
+            let playlist_check = client
+                .spotify
+                .playlist_check_follow(id.to_owned(), &[uid])
+                .await;
+            if playlist_check.is_err() {
+                anyhow::bail!("Could not find playlist {}", id.to_string())
+            }
+
+            // Won't delete if not following
+            let following = playlist_check.unwrap().pop();
+            if following.unwrap() {
```

Comment:
```suggestion
            let following = playlist_check.unwrap().pop().unwrap();
            if following {
```

can move the `uwnrap` upper.